uint8[] rightXY
uint8[] rightZ
uint8[] leftXY
uint8[] leftZ
